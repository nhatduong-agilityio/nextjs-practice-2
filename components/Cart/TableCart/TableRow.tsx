// Libs
import { CloseIcon } from '@chakra-ui/icons'
import { Box, Heading, HStack, IconButton, Td, Text, Tr } from '@chakra-ui/react'
import React, { useCallback, useState } from 'react'

// Components
import Image from '@components/Common/Image'

import SOFA_GREEN from '@public/images/sofa-green.jpg'
import Quantity from '@components/Common/Quantity'
import SelectOption from '@components/Common/SelectOption'
import { ColorMock } from '@mock/dataMock'

const CART = {
  id: 1,
  name: 'Green Slepper Sofa',
  category: 'Sofa',
  images: {
    image: SOFA_GREEN,
  },
  imageAlt: 'Picture of Full set Grey Sofa With Pillows',
  description: 'Modern home in city center in the heart of historic Los Angele',
  details: [
    { id: 1, content: 'High Quality Leathers' },
    { id: 2, content: 'High Quality Paints' },
    { id: 3, content: 'High Quality Woods' },
    { id: 4, content: 'Cheaps Price' },
  ],
  color: 'green',
  price: 120,
  discount: 0,
  sellCount: 400,
  rating: 5,
  quantity: 1,
}

const TableRow = () => {
  const [quantity, setQuantity] = useState(CART.quantity)

  const handleSumQuantity = useCallback(() => {
    setQuantity(quantity + 1)
  }, [quantity])

  const handleSubQuantity = useCallback(() => {
    if (quantity > 1) setQuantity(quantity - 1)
  }, [quantity])

  return (
    <Tr>
      <Td borderColor='#29293E' px={0} py='25px'>
        <HStack>
          <Box>
            <Image
              src={CART.images.image}
              alt={CART.imageAlt}
              width={['70px', '120px']}
              height={['70px', '120px']}
              variant='primary'
            />
          </Box>
          <Heading w='full' size={['xs', 'sm']} fontWeight={['Regular', 'Bold']}>
            {CART.name}
          </Heading>
        </HStack>
      </Td>

      <Td borderColor='#29293E' px={0} py='25px'>
        <Box>
          <Quantity
            quantity={quantity}
            onHandleSumQuantity={handleSumQuantity}
            onHandleSubQuantity={handleSubQuantity}
            disableLabel
          />
        </Box>
      </Td>
      <Td borderColor='#29293E' px={0} py='25px'>
        <Box>
          <SelectOption name='Color' placeholder={CART.color} options={ColorMock} disableLabel />
        </Box>
      </Td>
      <Td borderColor='#29293E' px={0} py='25px'>
        <Text size={['20', '30']} variant='large'>
          ${CART.price}
        </Text>
      </Td>
      <Td borderColor='#29293E' px={0} py='25px'>
        <Box display='flex' justifyContent='flex-end'>
          <IconButton
            variant='ghost'
            aria-label='Delete cart item'
            icon={<CloseIcon width='20px' height='20px' fill='red' color='red' />}
            minW='max-content'
            w='max-content'
          />
        </Box>
      </Td>
    </Tr>
  )
}

export default TableRow
