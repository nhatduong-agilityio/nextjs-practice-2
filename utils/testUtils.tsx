import { ChakraProvider } from '@chakra-ui/react'
import { AuthProvider } from '@store/providers/AuthProvider'
import { LoadingProvider } from '@store/providers/LoadingProvider'
import type { RenderOptions, RenderResult } from '@testing-library/react'
import { render } from '@testing-library/react'
import { customTheme } from '@themes/index'
import type { ReactElement } from 'react'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ChakraProvider theme={customTheme}>
      <LoadingProvider>
        <AuthProvider>{children}</AuthProvider>
      </LoadingProvider>
    </ChakraProvider>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
