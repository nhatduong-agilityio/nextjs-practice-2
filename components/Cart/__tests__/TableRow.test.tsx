// Utils
import { render } from '@utils/testUtils'

// Components
import { Table, Tbody } from '@chakra-ui/react'
import TableRow from '../TableCart/TableRow'

describe('TableRow render', () => {
  it('Should TableRow match Snapshot', () => {
    const { container } = render(
      <Table>
        <Tbody>
          <TableRow />
        </Tbody>
      </Table>,
    )

    expect(container).toMatchSnapshot()
  })
})
