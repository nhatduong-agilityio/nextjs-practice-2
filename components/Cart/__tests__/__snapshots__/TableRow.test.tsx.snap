// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow render Should TableRow match Snapshot 1`] = `
<div>
  <table
    class="chakra-table css-5605sr"
  >
    <tbody
      class="css-0"
    >
      <tr
        class="css-0"
      >
        <td
          class="css-uwq8cp"
        >
          <div
            class="chakra-stack css-1igwmid"
          >
            <div
              class="css-0"
            >
              <div
                class="css-1l85dhf"
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
                >
                  <img
                    alt="Picture of Full set Grey Sofa With Pillows"
                    data-nimg="fill"
                    decoding="async"
                    sizes="100vw"
                    src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
                  />
                </span>
              </div>
            </div>
            <h2
              class="chakra-heading css-irb7fl"
            >
              Green Slepper Sofa
            </h2>
          </div>
        </td>
        <td
          class="css-uwq8cp"
        >
          <div
            class="css-0"
          >
            <div
              class="chakra-stack css-vedre6"
            >
              <p
                class="chakra-text css-b1tfl1"
              >
                Quantity
              </p>
              <div
                class="chakra-stack css-15s55ey"
              >
                <button
                  class="chakra-button css-tum80v"
                  type="button"
                >
                  -
                </button>
                <div
                  class="css-lnxiq7"
                >
                  1
                </div>
                <button
                  class="chakra-button css-t9d6ji"
                  type="button"
                >
                  +
                </button>
              </div>
            </div>
          </div>
        </td>
        <td
          class="css-uwq8cp"
        >
          <div
            class="css-0"
          >
            <div
              class="chakra-stack css-vedre6"
              style="margin: 0px;"
            >
              <p
                class="chakra-text css-n9821e"
              >
                Color
              </p>
              <div
                class="css-1ki54i"
                style="margin: 0px;"
              >
                <div
                  class="chakra-select__wrapper css-1plvatr"
                >
                  <select
                    class="chakra-select css-5h0fr2"
                  >
                    <option
                      value=""
                    >
                      green
                    </option>
                    <option
                      value="blue"
                    >
                      Blue
                    </option>
                    <option
                      value="purple"
                    >
                      Purple
                    </option>
                    <option
                      value="red"
                    >
                      Red
                    </option>
                    <option
                      value="orange"
                    >
                      Orange
                    </option>
                    <option
                      value="yellow"
                    >
                      Yellow
                    </option>
                    <option
                      value="green"
                    >
                      Green
                    </option>
                  </select>
                  <div
                    class="chakra-select__icon-wrapper css-iohxn1"
                  >
                    <svg
                      aria-hidden="true"
                      class="chakra-select__icon"
                      focusable="false"
                      role="presentation"
                      style="width: 1em; height: 1em; color: currentColor;"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </td>
        <td
          class="css-uwq8cp"
        >
          <p
            class="chakra-text css-15fwflc"
          >
            $
            120
          </p>
        </td>
        <td
          class="css-uwq8cp"
        >
          <div
            class="css-1bvc4cc"
          >
            <button
              aria-label="Delete cart item"
              class="chakra-button css-fmejuy"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-1cta3o2"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
